### ISSUE TITLE
Report does not display for (older) projects

### ISSUE THREAD

**Issue by tinok** [kobotoolbox/kpi#1295]
For older projects, the reports do not load in kpi. 

1. https://kf.kobotoolbox.org/forms/#/forms/aGgKXX7VXHCewSbj3kSaea/data/report
1. https://kf.kobotoolbox.org/forms/#/forms/aDFftjXyJ3NttDx3bgbFDp/data/report
1. https://kf.kobotoolbox.org/forms/#/forms/ahiMczCjGLgg9Gq8bRPidS/data/report

The API call gets a 404 reponse

```
{
    "detail": "Not found."
}
```

---
Edit: updated urls with missing `data/`, this is not related to the issue

**Comment by magicznyleszek** [comment]
The error doesn't happen for form `aDFftjXyJ3NttDx3bgbFDp`, but still does for `ahiMczCjGLgg9Gq8bRPidS` - @tinok @jnm how do we want to proceed with this?

**Comment by tinok** [comment]
@magicznyleszek Thanks for checking on this ancient issue again... 

Yes I think it would be good to investigate why this is happening. If we know what the underlying cause is we can decide what to do about it at least.

**Comment by noliveleger** [comment]
@tinok :
1.  Asset `aGgKXX7VXHCewSbj3kSaea`: The error I got when I tried to load it when a JS error, not a 404. The `report_styles` and deployment data were wrong  (like malformed JSON - I don't know why). After reset it and redeploying it, everything works as expected. 
2. Asset `aDFftjXyJ3NttDx3bgbFDp`: Owner is `tinok4`, not `tinokreutzer` like the first one. `tinokreutzer` does not have any permission on this form, thus receives a 404 when accessing it. (FYI, it works perfectly fine with `tinok4` account)
3. Asset `ahiMczCjGLgg9Gq8bRPidS`: Works  as expected. Owner is `drc_14`. `tinokreutzer` has view access. 


When projects are transferred from legacy interface, only the projects owned by the user are transferred. If the user has view access on other projects in the legacy interface, other projects and their permissions must be transferred too.
